src/Servers/Kestrel/Transport.NamedPipes/src/Internal/NamedPipeConnectionListener.cs

```diff
@@ -40,6 +40,7 @@ internal sealed class NamedPipeConnectionListener : IConnectionListener
         _log = loggerFactory.CreateLogger("Microsoft.AspNetCore.Server.Kestrel.Transport.NamedPipes");
         _endpoint = endpoint;
         _options = options;
+        _namedPipeServerStreamPool = new DefaultObjectPoolProvider().Create(new NamedPipeServerStreamPoolPolicy(this));
```

Comment:
should we explicitly set `{ MaximumRetained = ... }`? or happy to just use defaults?

However: 👍 for choice:

``` txt
| Method |     Mean |    Error |   StdDev |   Gen0 | Allocated |
|------- |---------:|---------:|---------:|-------:|----------:|
|  Queue | 17.73 ns | 0.219 ns | 0.205 ns |      - |         - |
|  Stack | 22.39 ns | 0.139 ns | 0.123 ns | 0.0038 |      32 B |
|    Bag | 62.09 ns | 0.259 ns | 0.230 ns |      - |         - |
|   Pool | 16.31 ns | 0.232 ns | 0.194 ns |      - |         - |
```

Reply:
I think the default is fine for now. If we get reports or evidence that it's a problem then we can look at it more.